tests/Component.Tests/EventDispatcherTests.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Component.Tests.Fakes;
+using Components;
+using FluentAssertions;
+using Microsoft.Framework.DependencyInjection;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace Component.Tests
+{
+    public class EventDispatcherTests
+    {
+        static IServiceProvider CreateProvider() => new ServiceCollection().AddInMemoryEventStream().BuildServiceProvider();
+        static EventStream CreateEventStream(IServiceProvider serviceProvider = null) => (serviceProvider ?? CreateProvider()).GetRequiredService<EventStream>();
+        static EventDispatcher CreateDispatcher(IServiceProvider serviceProvider = null) => (serviceProvider ?? CreateProvider()).GetRequiredService<EventDispatcher>();
+
+        [Fact]
+        public static void GivenANullEntityThrowArgumentNullException()
```

Comment:
I don't particularly mind which one we go with. However, to me, Given is for setup, When is for action, Then is for assertion.

If you lack a When, nothing has happened?